satpy/readers/__init__.py

```diff
@@ -420,40 +422,125 @@ def group_files(files_to_sort, reader=None, time_threshold=10,
         a `Scene` object.
 
     """
-    # FUTURE: Find the best reader for each filename using `find_files_and_readers`
-    if reader is None:
-        raise ValueError("'reader' keyword argument is required.")
-    elif not isinstance(reader, (list, tuple)):
+
+    if reader is not None and not isinstance(reader, (list, tuple)):
         reader = [reader]
 
-    # FUTURE: Handle multiple readers
-    reader = reader[0]
-    reader_configs = list(configs_for_reader(reader, ppp_config_dir))[0]
     reader_kwargs = reader_kwargs or {}
-    try:
-        reader_instance = load_reader(reader_configs, **reader_kwargs)
-    except (KeyError, IOError, yaml.YAMLError) as err:
-        LOG.info('Cannot use %s', str(reader_configs))
-        LOG.debug(str(err))
-        # if reader and (isinstance(reader, str) or len(reader) == 1):
-        #     # if it is a single reader then give a more usable error
-        #     raise
-        raise
-
-    if group_keys is None:
-        group_keys = reader_instance.info.get('group_keys', ('start_time',))
-    file_keys = []
-    # make a copy because filename_items_for_filetype will modify inplace
+
+    reader_files = _assign_files_to_readers(
+            files_to_sort, reader, ppp_config_dir, reader_kwargs)
+
+    if reader is None:
+        reader = reader_files.keys()
+
+    file_keys = _get_file_keys_for_reader_files(
+            reader_files, group_keys=group_keys)
+
+    file_groups = _get_sorted_file_groups(file_keys, time_threshold)
+
+    return [{rn: file_groups[group_key].get(rn, []) for rn in reader} for group_key in file_groups]
+
+
+def _assign_files_to_readers(files_to_sort, reader_names, ppp_config_dir,
+                             reader_kwargs):
+    """Assign files to readers.
+
+    Given a list of file names (paths), match those to reader instances.
+
+    Internal helper for group_files.
+
+    Args:
+        files_to_sort (Collection[str]): Files to assign to readers.
+        reader_names (Collection[str]): Readers to consider
+        ppp_config_dir (str):
+        reader_kwargs (Mapping):
+
+    Returns:
+        Mapping[str, Tuple[reader, Set[str]]]
+        Mapping where the keys are reader names and the values are tuples of
+        (reader_configs, filenames).
+    """
+
     files_to_sort = set(files_to_sort)
-    for _, filetype_info in reader_instance.sorted_filetype_items():
-        for f, file_info in reader_instance.filename_items_for_filetype(files_to_sort, filetype_info):
-            group_key = tuple(file_info.get(k) for k in group_keys)
-            file_keys.append((group_key, f))
+    D = {}
```

Comment:
Can you use something other than a single character (and lowercase it)?

Reply:
Changed to `reader_dict`.